src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +148,69 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust: { node: Node<NodeData>; zIndex: number }[] = [];
     // set the zIndex of all edges
     for (const e of edges) {
         let zIndex = Math.max(
             byId.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
             byId.get(e.target)?.zIndex ?? STARTING_Z_INDEX
         );
 
+        const connectedNodes = nodes.filter((n) => n.id === e.source || n.id === e.target);
+
         if (e.selected && zIndex < MIN_SELECTED_INDEX) {
             zIndex += SELECTED_ADD;
+            // If an edge is selected, make the connected nodes the same zIndex
+            connectedNodes.forEach((n) => {
+                if (n.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.push({ node: n, zIndex });
+                }
+            });
+        } else if (connectedNodes.some((n) => n.selected)) {
+            // If an edge is connectec to a selected node, we need to make the other node it is connected to the same zIndex
+            connectedNodes.forEach((n) => {
```

Comment:
I don't understand this and why you only update the zIndexes of nodes that are selected or that are connected to a selected edge. Why do you not update the zIndexes of both nodes connected to this edge?

Reply:
The entirety of this code was wack-a-mole esque "solve one problem and three more appear" so everything i did here was because changing the z index of one thing caused it to be over something else that needed to be raised, etc. If you were to try this yourself i bet you'd end up with the same kind of logic for raising up what is connected.